rclcpp/test/rclcpp/CMakeLists.txt

```diff
@@ -463,6 +479,14 @@ if(TARGET test_type_support)
   )
   target_link_libraries(test_type_support ${PROJECT_NAME})
 endif()
+ament_add_gmock(test_typesupport_helpers generic/test_typesupport_helpers.cpp)
+if(TARGET test_typesupport_helpers)
+  ament_target_dependencies(test_typesupport_helpers
+    "rcpputils"
+    "rosidl_runtime_cpp"
+  )
```

Comment:
I actually think this statement isn't needed since you also link against `rclcpp` itself? Same for the other test block above.

Reply:
bump

Reply:
The `ament_target_dependencies` or the `target_link_libraries`? Lots of other packages also have "rosidl_runtime_cpp" in `ament_target_dependencies` and link against `rclcpp`, should they be changed too?

Reply:
The `ament_target_dependencies`.

> Lots of other packages also have "rosidl_runtime_cpp" in ament_target_dependencies and link against rclcpp, should they be changed too?

I'm not sure, but you could try it on yours first. This isn't super important, I just don't see why you need it. It could be that the others are redundant too.

Reply:
I am pretty certain that only the `target_link_libraries` line would suffice.

Reply:
Yep, it works. Should I remove all those `ament_target_dependencies` that are exported dependencies of rclcpp?

Reply:
I'd just do it for your entries for now. Keep the pr scoped. I'll consider updating the others later.